landoapi/landing_worker.py

```diff
@@ -69,6 +72,53 @@ def __init__(self, sleep_seconds=5):
         signal.signal(signal.SIGINT, self.exit_gracefully)
         signal.signal(signal.SIGTERM, self.exit_gracefully)
 
+    def _setup_ssh(self):
+        """Fetch a private SSH key from the environment and add it to ssh-agent.
+        """
+        SSH_PRIVATE_KEY_ENV_KEY = "SSH_PRIVATE_KEY"
+
+        # Fetch ssh private key from the environment. Note that this key should be
+        # stored in standard format including all new lines and new line at the end
+        # of the file.
+        ssh_private_key = os.environ.get(SSH_PRIVATE_KEY_ENV_KEY)
+        if not ssh_private_key:
+            logger.warning(f"No {SSH_PRIVATE_KEY_ENV_KEY} present in environment.")
+            return
+
+        # Set all the correct environment variables
+        agent_process = subprocess.run(
+            ["ssh-agent", "-s"], capture_output=True, universal_newlines=True
+        )
+
+        # This pattern will match keys and values, and ignore everything after the
+        # semicolon. For example, the output of `agent_process` is of the form:
+        #     SSH_AUTH_SOCK=/tmp/ssh-c850kLXXOS5e/agent.120801; export SSH_AUTH_SOCK;
+        #     SSH_AGENT_PID=120802; export SSH_AGENT_PID;
+        #     echo Agent pid 120802;
+
+        pattern = re.compile("(.+)=([^;]*)")
+        for key, value in pattern.findall(agent_process.stdout):
+            self.config["SSH_ENV_KEYS"].append(key)
+            os.environ[key] = value
+
+        # Add private SSH key to agent
+        # NOTE: ssh-add seems to output everything to stderr, including upon exit 0.
+        add_process = subprocess.run(
+            ["ssh-add", "-"],
+            input=ssh_private_key,
+            capture_output=True,
+            universal_newlines=True,
+        )
+        if add_process.returncode != 0:
+            raise Exception(add_process.stderr)
+        logger.info("Added private SSH key from environment.")
+
+    def _teardown_ssh(self):
+        """Clean up relevant environment variables we've set up for ssh-agent.
+        """
+        for key in self.config["SSH_ENV_KEYS"]:
+            del os.environ[key]
```

Comment:
Can you explain why we need this?

Reply:
Good point. I'm not really sure we need this, it doesn't really do anything. If the worker is killed or dies, then these variables will be reset anyway on the next start. Will remove.